DogScepterLib/Core/GMData.cs

```diff
@@ -115,153 +201,160 @@ public class GMStats
 
-                    for (int i = StringCache.Count; i < list.Count; i++)
-                        if (list[i].Content == content)
-                            return list[i];
-                }
-                else
+                for (int i = StringCache.Count; i < list.Count; i++)
                 {
-                    foreach (GMString str in list)
-                    {
-                        if (str.Content == content)
-                            return str;
-                    }
+                    if (list[i].Content == content)
+                        return list[i];
                 }
-
-                GMString res = new();
-                res.Content = content;
-                list.Add(res);
-                return res;
             }
+            else
+            {
+                foreach (GMString str in list.Where(str => str.Content == content))
+                    return str;
+            }
+
+            GMString res = new();
+            res.Content = content;
+            list.Add(res);
+            return res;
         }
+    }
 
-        public GMString DefineString(string content, out int index)
+    public GMString DefineString(string content, out int index)
+    {
+        if (content == null)
         {
-            if (content == null)
-            {
-                index = -1;
-                return null;
-            }
+            index = -1;
+            return null;
+        }
 
-            var list = ((GMChunkSTRG)Chunks["STRG"]).List;
+        GMPointerList<GMString> list = ((GMChunkSTRG)Chunks["STRG"]).List;
 
-            lock (list)
+        lock (list)
+        {
+            if (StringCache != null)
             {
-                if (StringCache != null)
+                if (StringCache.TryGetValue(content, out int stringIndex))
                 {
-                    if (StringCache.TryGetValue(content, out int stringIndex))
-                    {
-                        index = stringIndex;
-                        return list[stringIndex];
-                    }
-
-                    for (int i = StringCache.Count; i < list.Count; i++)
-                    {
-                        if (list[i].Content == content)
-                        {
-                            index = i;
-                            return list[i];
-                        }
-                    }
+                    index = stringIndex;
+                    return list[stringIndex];
                 }
-                else
+
+                for (int i = StringCache.Count; i < list.Count; i++)
                 {
-                    for (int i = 0; i < list.Count; i++)
-                    {
-                        if (list[i].Content == content)
-                        {
-                            index = i;
-                            return list[i];
-                        }
-                    }
+                    if (list[i].Content != content) continue;
```

Comment:
Hmm... Not sure if I'm a big fan of changing this to use `continue`. It doesn't get rid of anything, and arguably makes it harder to read.

Reply:
Whether it makes it harder to read or not, is dependant on the person, I guess. For me it makes it easier to read, due to also reducing nesting. Just a standard "return early" pattern.

Reply:
The way I see it, this just adds an extra control flow keyword and sort of masks the intent of the comparison. It's searching for a string that has the specified content, not to _ignore_ all the strings that don't have the same content, if you get what I mean. Usually "return early" patterns make more sense when there's more than one condition.

Reply:
> It's searching for a string that has the specified content, not to ignore all the strings that don't have the same content,
but those are the exact same conditions though.